components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -1986,6 +2033,142 @@ func (backend *ES2Backend) getSearchResult(reportId string, filters map[string][
+func (backend *ES2Backend) getControlSummaryFromControlIndex(ctx context.Context, controlId []string, filters map[string][]string, esIndex string, size int32) (map[string]*reportingapi.ControlSummary, error) {
+	nodeStatus := "nodes.status"
+	controlSummaryMap := make(map[string]*reportingapi.ControlSummary)
+	client, err := backend.ES2Client()
+	if err != nil {
+		return nil, err
+	}
+	boolQuery := getControlSummaryFilters(controlId, filters)
+
+	filterQuery := elastic.NewBoolQuery()
+
+	searchSource := elastic.NewSearchSource().
+		Query(boolQuery).
+		FetchSource(false).Size(1)
+
+	passedFilter := elastic.NewFilterAggregation().Filter(elastic.NewBoolQuery().
+		Must(elastic.NewTermQuery(nodeStatus, "passed")))
+
+	failedFilter := elastic.NewFilterAggregation().Filter(elastic.NewBoolQuery().
+		Must(elastic.NewTermQuery(nodeStatus, "failed")))
+
+	skippedFilter := elastic.NewFilterAggregation().Filter(elastic.NewBoolQuery().
+		Must(elastic.NewTermQuery(nodeStatus, "skipped")))
+
+	setFlags, err := filterQueryChange(firstOrEmpty(filters["end_time"]), firstOrEmpty(filters["start_time"]))
+	if err != nil {
+		logrus.Errorf("Unable to fetch details for flags %v", err)
+	}
+	for _, flag := range setFlags {
+		termQuery := elastic.NewTermsQuery("nodes."+flag, true)
+		filterQuery = filterQuery.Must(termQuery)
+	}
+
+	filterAggregationForLatestRecord := elastic.NewFilterAggregation().Filter(filterQuery).
+		SubAggregation("failed", failedFilter).
+		SubAggregation("passed", passedFilter).
+		SubAggregation("skipped", skippedFilter)
+
+	subAggregationNodes := elastic.NewNestedAggregation().Path("nodes").SubAggregation("status", filterAggregationForLatestRecord)
+
+	//waivedFilter := elastic.NewFilterAggregation().Filter(waivedQuery)
+
+	controlTermsAggregation := elastic.NewTermsAggregation().Field("control_id").SubAggregation("nodes", subAggregationNodes).Size(int(size))
+
+	searchSource.Aggregation("controls", controlTermsAggregation)
+
+	source, err := searchSource.Source()
+	if err != nil {
+		return nil, errors.Wrapf(err, "%s unable to get Source", "ControlSummary")
+	}
+	LogQueryPartMin(esIndex, source, fmt.Sprintf("%s query", "ControlSummary"))
+
+	searchResult, err := client.Search().
+		SearchSource(searchSource).
+		Index(esIndex).
+		Do(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if controlBuckets, found := searchResult.Aggregations.Terms("controls"); found && len(controlBuckets.Buckets) > 0 {
+		for _, bucket := range controlBuckets.Buckets {
+			controlSummary := &reportingapi.ControlSummary{
+				Passed:  &reportingapi.Total{},
+				Skipped: &reportingapi.Total{},
+				Failed:  &reportingapi.Failed{},
+				Waived:  &reportingapi.Total{},
+			}
+			if nodesBuckets, found := bucket.Aggregations.Nested("nodes"); found {
+				controlSummary = getControlSummaryResult(nodesBuckets)
+			}
+
+			id, _ := bucket.Key.(string)
```

Comment:
we should not omit the second value in type assertion to see if the assertion is correct or not. If we are sure that there won't be no other value apart from string, we can directly write 
id := bucket.Key.(string)
instead of these, how about using `KeyAsString`?

Reply:
keyAsString, doesn't gives us a flag if the key is found or not in the result. So this is necessary. I have added the flag to check if key is present or not.